docs/extending-typespec/linters.md

```diff
@@ -119,24 +134,26 @@ First you'll want to create an instance of the rule tester using `createRuleTest
 You can then provide different test checking the rule pass or fails.
 
 ```ts
-import { createRuleTester } from "@typespec/lint/testing";
-import { noFooModelRule } from "./no-foo-model.js";
+import { createRuleTester } from "@typespec/compiler/testing";
+import { requiredDocRule } from "./rules/required-doc.rule.js";
 
-let ruleTester: RuleTester;
+describe("required-doc rule", () => {
+  let ruleTester: RuleTester;
 
-beforeEach(() => {
-  const runner = createTestRunner();
-  ruleTester = createRuleTester(runner, noFooModelRule);
-});
+  beforeEach(() => {
+    const runner = createTestRunner();
+    ruleTester = createRuleTester(runner, requiredDocRule, "@typespec/my-linter");
```

Comment:
This differ from the linter library because now as the emitted diagnostic is fully configured within the rule there is no ref of the library anymore. (While before it was just reporting a std diagnostic).

This makes it that if we want the test to be able to check the resolved diagnostic we need to explicitly pass the library name.

There is a few options here:
1. Have what we have here, a little more extra config
2. Remove that parameter(or make it optional) and then the diagnostic in the test will not be prefixed by the lbirary name. Maybe just have a placeholder `code: "{library-name}:no-foo-model",`
3. The `toEmitDiagnostics` actually doesn't expect the resolve diagnostic but just what the rule pass  so you'd have that below
```
ruleTester.expect(`model Foo {}`).toEmitDiagnostics({
    messageId: "default"
});
```
Disadantage of this is we don't get to test message interpolation